### ISSUE TITLE
Align hyperparameter tuning GBT models

### ISSUE THREAD

**Issue by fritshermans** [INRIA/scikit-learn-mooc#835]
In the section 'Hyperparameter tuning by randomized search', different hyperparameters are tuned for Histogram gradient-boosting decision trees than in section 'Hyperparameter tuning with ensemble models'. In the former section, `l2_regularization` and `max_bins` are tuned but not in the latter. In the latter section `max_depth` is tuned but not in the former section. My proposal would be to:

- remove tuning of `max_bins`; this argument is only to set the granularity of optimal split finding in the trees so I don't think it affects the complexity of the model and the ability to generalize
- add a line on how l2-regularisation works for GBT as it is not explained or remove it
- add tuning of `max_depth` in the former section

Please let me know what you think of this. I would be happy to create a PR.
